google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/javadsl/GCStorage.scala

```diff
@@ -233,6 +234,38 @@ object GCStorage {
                    contentType: ContentType): Source[StorageObject, NotUsed] =
     GCStorageStream.putObject(bucket, objectName, data.asScala, contentType.asInstanceOf[ScalaContentType]).asJava
 
+  /**
+   * Uploads object by making multiple requests
+   *
+   * @see https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload
+   *
+   * @param bucket the bucket name
+   * @param objectName the object name
+   * @param contentType `ContentType`
+   * @param chunkSize the size of the request sent to google cloud storage in bytes, must be a multiple of 256KB
+   * @param metadata custom metadata for the object
+   * @return a `Sink` that accepts `ByteString`'s and materializes to a `Future` of `StorageObject`
+   */
+  def resumableUpload(bucket: String,
+                      objectName: String,
+                      contentType: ContentType,
+                      chunkSize: java.lang.Integer,
+                      metadata: java.util.Map[String, String]): Sink[ByteString, CompletionStage[StorageObject]] = {
```

Comment:
I think the extra overloads are fine. I was considering whether we should mark the other as deprecated, but considering how this has only been addressed now there are probably many users not concerned with metadata.